riocli/device/report.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2024 Rapyuta Robotics
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta
+
+import click
+from click_help_colors import HelpColorsCommand
+from rapyuta_io.clients import SharedURL
+
+from riocli.config import new_client
+from riocli.constants import Colors, Symbols
+from riocli.device.util import name_to_guid, report_device_api_call
+from riocli.utils.spinner import with_spinner
+
+
+@click.command(
+    "report",
+    cls=HelpColorsCommand,
+    help_headers_color=Colors.YELLOW,
+    help_options_color=Colors.GREEN,
+)
+@click.option(
+    "--force", "-f", "--silent", "force", is_flag=True, help="Skip confirmation"
+)
+@click.option(
+    "--share",
+    "-s",
+    is_flag=True,
+    help="Share the device logs with Rapyuta support",
+    default=False,
+)
+@click.option(
+    "--expiry",
+    "-e",
+    help="Expiry time for the shared URL in days",
+    default=7,
+    required=False,
+    show_default=True,
+)
+@click.argument("device-name", type=str)
+@name_to_guid
+@with_spinner(text="Upload debugging files...")
+def report_device(
+    device_name: str,
+    device_guid: str,
+    force: bool,
+    share: bool,
+    expiry: int,
+    spinner=None,
+) -> None:
+    """
+    Report a device and get its debug logs in uploads section.
+    """
+    if not force:
+        with spinner.hidden():
+            click.confirm(
+                "Report Device?",
+                abort=True,
+            )
+    try:
+        # Upload the debug logs
+        response = report_device_api_call(device_guid=device_guid)
+
+        if share:
+            spinner.text = click.style("Creating shared URL...", fg=Colors.YELLOW)
+            public_url = shared_url_debug_logs(
+                device_guid, response["response"]["data"]["request_uuid"], expiry
+            )
+            click.secho(public_url.url, fg=Colors.GREEN)
+
+        spinner.text = click.style("Device reported successfully.", fg=Colors.GREEN)
+        spinner.green.ok(Symbols.SUCCESS)
+
+    except Exception as e:
+        spinner.text = click.style(
+            "Failed to report device: {}".format(e), fg=Colors.RED
+        )
+        spinner.red.fail(Symbols.ERROR)
+
+
+@with_spinner(text="Creating shared URL...")
+def shared_url_debug_logs(device_guid: str, request_id: str, expiry: int, spinner=None):
```

Comment:
todo: Move this to util.
```suggestion
def generate_shared_url(device_guid: str, request_id: str, expiry: int, spinner=None):
```